src/Device.vala

```diff
@@ -97,14 +125,88 @@ public class Power.Device : Object {
 
             update_properties ();
             upower_device.g_properties_changed.connect (update_properties);
+
+            update_description ();
+            notify["percentage"].connect (update_description);
         } catch (IOError e) {
             critical (e.message);
         }
     }
 
     private void update_properties () {
+        coarse_battery_level = upower_device.battery_level != 1;
+
+        model = upower_device.model; // Can sometimes update eg when phone is trusted
         percentage = upower_device.percentage;
```

Comment:
We should probably check for coarse battery level here and if we can get a coarse battery level we should set percentage to reflect that level since percentage can be inaccurate in those cases